src/encoder.rs

```diff
@@ -543,6 +543,19 @@ pub struct DeblockState {
   pub block_delta_multi: bool,
 }
 
+impl DeblockState {
+  pub fn new(config: &EncoderConfig) -> Self {
+    let mut state = DeblockState { ..Default::default() };
+    for level in &mut state.levels {
```

Comment:
Since rav1e already codes keyframes very well for video and spends a lot of bits on it, it would be preferably to make the deblocking strength control only available to non intra refreshes. However, keeping it available for still picture mode is still advisable since lowering deblocking strength seems to provide higher fidelity results overall.

Here is how I would implement it:

if still_picture {

do deblocking_strength control }

else if !PRIMARY_REF_NONE(don't exactly know how to set this) {
do deblocking_strength control }

That way, we can actually get the best of both worlds.

Reply:
That works for now. Thank you.